### PR TITLE
Update test runners to Fedora 39

### PR DESCRIPTION
**Pull request by achilleas-k**
- Use Fedora 39 to run tests.
- Update test examples.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 - `./cmd/build` takes a config file as argument to build an image.  For example:
 ```
 go build -o bin/build ./cmd/build
-sudo ./bin/build -output ./buildtest -rpmmd /tmp/rpmmd -distro fedora-38 -image qcow2 -config test/configs/embed-containers.json
+sudo ./bin/build --output ./buildtest --rpmmd /tmp/rpmmd --distro fedora-39 --image qcow2 --config test/configs/embed-containers.json
```
Comment (mvo5) [inline test/README.md:8]: Thanks for this!

**Review by mvo5 — DISMISSED** [review]
This looks fine

**Review by thozza — APPROVED** [review]
LGTM

**Comment by achilleas-k** [comment]
Something odd happening with the merge queue.  Investigating.

**Comment by achilleas-k** [comment]
Found an issue with building while looking into this (see #367).  Not sure what the cause of the merge-queue failure was but it might be related to all the rebuilds that were happening because of it.

**Comment by achilleas-k** [comment]
More info: Flakiness + retries don't mix well with the merge-queue.  For example, right now the CI in gitlab is technically in progress still without a failure, but there is a build that failed with:
```
[ERROR] depsolve failed: DNF error occurred: RepoError: There was a problem reading a repository: Failed to download metadata for repo 'a696f78f8fe604ad238f1034c062b09c7a9ddb6d2016a83944e0f0685ab84e9f' [appstream: https://rpmrepo.osbuild.org/v2/mirror/rhvpn/el8/el8-x86_64-appstream-r8.5-20230809]: Yum repo downloading error: Downloading error(s): repodata/b591d9fa12e0ba5c6d75b84b605f44c1daefd514068152a3fd8ba53fdb5975eb-primary.xml.gz - Cannot download, all mirrors were already tried without success; repodata/689bd45a22fedbffbc3e3b9d82109a2c6ed02ee8b2962136d63a3026f24d6ef0-filelists.xml.gz - Cannot download, all mirrors were already tried without success; repodata/167822ee59a5cbd329d31b2ff030cb2146b0fd624e390ffe3ac98fb641fb62f4-comps-AppStream.x86_64.xml.gz - Cannot download, all mirrors were already tried without success; repodata/d12f84e558ef647914eb6d20f034920765390d579888595800f915a133457a22-modules.yaml.gz - Cannot download, all mirrors were already tried without success
```

This is now being retried and will most likely succeed.

Whenever a test fails, before it's retried, it updates the test status on GitHub to failed.  If this happens on the merge-queue, the PR is removed from the queue and if the retry succeeds after that, you get into the interesting situation where:
- The CI run in GitLab is green (after retries) but,
- GitHub says it removed the PR from the queue because of a failure.

I think we can solve this by removing the `update_github_status` run from each pipeline and defining a new job with `when: on_failure` to update the status only when everything has been retried and is still failing.

**Pull request merged**